frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -1,129 +1,272 @@
+              else -> null
+            }
           }
+    } catch (e: Exception) {
+      LOG.warn("Failed to load unmerged branches for root: $root", e)
     }
+
+    return result
   }
 
-  private fun validateBranchName(): ValidationInfoBuilder.(JTextField) -> ValidationInfo? = {
-    val insertedText = it.text
+  private fun validateBranchName(): ValidationInfo? {
+    val insertedText = branchField.getText()
+
+    if (insertedText.isNullOrEmpty()) {
+      return ValidationInfo(
+        getString(
+          "action.GitMachete.BaseSlideInBranchBelowAction.dialog.slide-in.no-branch-selected"
+        ),
+        branchField
+      )
+    }
+
     val errorInfo = git4idea.validators.checkRefName(insertedText)
-    if (errorInfo != null) error(errorInfo.message)
-    else if (insertedText == parentName) {
-      error(
+    if (errorInfo != null) {
+      return ValidationInfo(errorInfo.message, branchField)
+    } else if (insertedText == parentName) {
+      return ValidationInfo(
         getString(
           "action.GitMachete.BaseSlideInBranchBelowAction.dialog.slide-in.error.slide-in-under-itself"
-        )
+        ),
+        branchField
       )
     } else {
       val entryByName = branchLayout.findEntryByName(insertedText)
       if (entryByName.map(isDescendantOf(presumedDescendantName = parentName)).getOrElse(false)) {
-        error(
+        return ValidationInfo(
           getString(
             "action.GitMachete.BaseSlideInBranchBelowAction.dialog.slide-in.error.slide-in-under-its-descendant"
-          )
+          ),
+          branchField
         )
       } else {
         if (insertedText in rootNames) { // the provided branch name refers to the root entry
-          reattachCheckbox?.isEnabled = false
-          reattachCheckbox?.isSelected = true
+          reattachCheckbox.isEnabled = false
+          reattachCheckbox.isSelected = true
         } else {
           val existsAndHasAChild = entryByName.orNull?.children?.nonEmpty() ?: false
-          reattachCheckbox?.isEnabled = existsAndHasAChild
-          reattachCheckbox?.isSelected =
-            (reattachCheckbox?.isSelected ?: false) && existsAndHasAChild
+          reattachCheckbox.isEnabled = existsAndHasAChild
+          reattachCheckbox.isSelected = reattachCheckbox.isSelected && existsAndHasAChild
         }
-
-        null
       }
     }
+    return null
   }
 
-  private fun Cell<JBTextField>.startTrackingValidationIfNeeded() {
-    if (branchName.isEmpty()) {
-      component.document.addDocumentListener(
-        object : DocumentAdapter() {
-          override fun textChanged(e: DocumentEvent) {
-            startTrackingValidation()
-            component.document.removeDocumentListener(this)
-          }
-        }
+  private fun updateBranchesField() {
+    val branches =
+      GitBranchUtil.sortBranchNames(gitRepository.branches.localBranches.map { it.name })
+        .filter { it != parentName }
```

Comment:
Fix for https://github.com/VirtusLab/git-machete-intellij-plugin/issues/977. I am closing the issue. I ve got one thing to be resolved here so please **hold on with the further review for now**.

Reply:
ready for rereview

Reply:
Lemme install it locally, as per _tradition_